docs: Document new TLS functionality in the Helm chart (#1645)
redpanda-data/documentation: docs/manage/kubernetes/security/kubernetes-tls.mdx

</details>

#### Test external connections

> To test external connections, you need to enable external access using a custom domain. The SAN list of your self-signed certificate does not contain the IP addresses of your worker nodes, but when you enable external access using a custom domain, that domain is included in the SAN list. Then, you can use rpk on your local machine to communicate with the cluster externally using the self-signed certificate for encryption.

Reviewer: ```suggestion
To test external connections, you must enable external access using a custom domain. The SAN list of your self-signed certificate does not contain the IP addresses of your worker nodes, but when you enable external access using a custom domain, that domain is included in the SAN list. Then, you can use rpk on your local machine to communicate with the cluster externally using the self-signed certificate for encryption.
```

kubectl exec redpanda-0 -n redpanda -c redpanda -- rpk cluster info --brokers <subdomain>.<custom-domain>:<kafka-api-port>
```

<details>
<summary>
Example output
</summary>

```
CLUSTER
=======
redpanda.19ae8532-c8fa-49ed-8b35-82d74813db3a

BROKERS
=======
ID    HOST                                             PORT
0*    redpanda-0.redpanda.redpanda.svc.cluster.local.  9093
1     redpanda-1.redpanda.redpanda.svc.cluster.local.  9093
2     redpanda-2.redpanda.redpanda.svc.cluster.local.  9093
```

</details>

#### Test external connections

Validate your external connection to Redpanda with rpk:

Replace the following placeholders:

- `<subdomain>`: The subdomain that's in the advertised address of one of your Redpanda brokers.
- `<custom-domain>`: Your domain.
- `<external-port>`: The port on which your cluster's Kafka API is exposed.

```bash
rpk cluster info --brokers <subdomain>.<custom-domain>:<external-port>
```

## Use Secrets to manage TLS certificates without cert-manager

If you prefer not to use cert-manager for TLS certificate management, you can manage your TLS configurations using Kubernetes Secrets. The following steps outline how to configure TLS with Secrets:

> 1. Create a Kubernetes Secret that contains your TLS certificates.

Reviewer: Wondering--why is "Secret" capitalized? When I wrote about this at Confluent, we always used lowercase: https://docs.confluent.io/platform/current/security/secrets.html
Unless it is a product or licensed feature, I'd say use lowercase. We don't have specific guidance on this in our Style Guide, but I'm thinking we should.

Author: It's a convention in Kubernetes to capitalize resource names. We call this out here: https://docs.redpanda.com/docs/deploy/deployment-option/self-hosted/kubernetes/kubernetes-production-deployment/#documentation-conventions

    Replace the `<path>` placeholders with the paths to your certificate files.

    - To create a Secret that includes your `tls.crt` and `tls.key` files:

      ```bash
      kubectl create secret tls <secret-name> \
        --cert=<path>tls.crt \
        --key=<path>tls.key \
        -n redpanda
      ```

>     - To include the `ca.crt` file, you need to create a generic secret:

Reviewer: ```suggestion
    - To include the `ca.crt` file, you must create a generic secret:
```

You can validate your internal connection to Redpanda with rpk by executing the following command.

```bash
kubectl exec redpanda-0 -n redpanda -c redpanda -- rpk cluster info \
  --tls-enabled \
  --brokers <broker-url>:<kafka-api-port>
```

If you are using a private CA, include the path to your CA certificate in the `--tls-truststore` option.

```bash
kubectl exec redpanda-0 -n redpanda -c redpanda -- rpk cluster info \
  --tls-enabled \
  --brokers <broker-url>:<kafka-api-port> \
  --tls-truststore <path-to-ca-certificate>
```

### Test external connections

To test external connections, you need to enable external access using a custom domain. You can use rpk on your local machine to communicate with the cluster externally using the TLS certificates for encryption.

```bash
rpk cluster info \
  --brokers <subdomain>.<custom-domain>:<external-kafka-api-port>
```

If you are using a private CA, include the path to your CA certificate in the `--tls-truststore` option.

```bash
rpk cluster info \
  --brokers <subdomain>.<custom-domain>:<external-kafka-api-port> \
  --tls-truststore <path-to-ca-certificate>
```

## Use multiple certificates for different listeners

> The Redpanda cluster provides granular control over the TLS certificates used by different listeners. This level of flexibility allows you ensure the required level of security for each listener. For example, you can use a self-signed certificate for the internal RPC listener, while using a Certificate Authority (CA) certificate for other listeners such as the Kafka API.

Reviewer: ```suggestion
The Redpanda cluster provides granular control over the TLS certificates used by different listeners. This level of flexibility enables you to ensure the required level of security for each listener. For example, you can use a self-signed certificate for the internal RPC listener, while using a Certificate Authority (CA) certificate for other listeners such as the Kafka API.
```

> By default, all listeners in Redpanda use the self-signed certificate defined globally in the `tls.certs.default` configuration. To customize the certificates for each listener, you can edit the `listeners.<listener-name>.tls.cert` setting

Reviewer: ```suggestion
By default, all listeners in Redpanda use the self-signed certificate defined globally in the `tls.certs.default` configuration. To customize the certificates for each listener, you can edit the `listeners.<listener-name>.tls.cert` setting.
```

    --tls-truststore ca.crt
  ```

> ### Use a CA certificate

Reviewer 2: Technically, the self-signed also uses a CA certificate, it's just a self-signed one. The generated CA can be downloaded and used by external clients to validate their connections.

> ### Use a CA certificate

Author: ```suggestion
### Use a public CA certificate
```
